### ISSUE TITLE
Segmenttion fault when syncing from scratch with 1.4.0.0

### ISSUE THREAD

**Issue by foomip** [decenomy/SAPP#19]
When syncing from scratch with sapphired on version 1.4.0.0 the daemon crashes with the following error

```
Segmentation fault (core dumped)
```

If I restart the daemon it keeps syncing and then eventually crashes again.

### Machine specs:
- OS: Ubuntu 18.0.4.4
- CPU: Ryzen 7 3700X
- RAM: 64GB
- Disk size: 1TB
- Disk Type (HD/SDD): HD (Raid 1)

**Comment by foomip** [comment]
Some extra info. Even after sync has completed the node is still failing with the segmentation fault.

Some sample logs
```
Added time data, samples 70, offset -10 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
socket recv error Connection reset by peer (104)
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901814, us=176.9.3.85:45328, peer=120
Added time data, samples 71, offset +0 (+0 minutes)
nTimeOffset = +0
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /Sapphire Core:1.4.0/: version 70927, blocks=206444, us=176.9.3.85:45328, peer=121
Added time data, samples 72, offset +0 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901814, us=176.9.3.85:45328, peer=122
Added time data, samples 73, offset +0 (+0 minutes)
nTimeOffset = +0
asking peer for sporks
CheckBlock : skipping transaction locking checks
UpdateTip: new best=23abddeaf98d5627dec33654e34ec9dd8090b91752453d8ee0f6c78c45b58609  height=901815 version=7  log2_work=68.7403139180786269  tx=10281823  date=2021-01-10 00:29:54 progress=1.000000  cache=0.4MiB(2075txo)
ERROR: ProcessBlock: Active Masternode not initialized.
ProcessNewBlock : ACCEPTED Block 901815 in 22 milliseconds with size=43938
AdvertiseLocal: advertising address 176.9.3.85:45328
socket recv error Connection reset by peer (104)
receive version message: /SAPP Core:1.3.3.1/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=123
Added time data, samples 74, offset +13 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
ProcessMessages: advertising address 176.9.3.85:45328
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:35672, peer=124
Added time data, samples 75, offset +0 (+0 minutes)
nTimeOffset = +0
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /Sapphire Core:1.4.0/: version 70927, blocks=901814, us=176.9.3.85:45328, peer=126
Added time data, samples 76, offset +0 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=127
Added time data, samples 77, offset +0 (+0 minutes)
nTimeOffset = +0
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /Sapphire Core:1.4.0/: version 70927, blocks=901815, us=176.9.3.85:45328, peer=128
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=129
Added time data, samples 78, offset -1 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=130
Added time data, samples 79, offset +0 (+0 minutes)
nTimeOffset = +0
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
connect() to 185.220.100.248:45328 failed after select(): Connection refused (111)
connect() to [2a01:4f9:2a:2443::50f]:45328 failed after select(): Connection refused (111)
connect() to [2a01:4f8:200:64c5::4b]:45328 failed after select(): Connection refused (111)
ProcessMessages: advertising address 176.9.3.85:45328
receive version message: /Sap Core:1.3.3/: version 70926, blocks=901815, us=[2a01:4f8:141:332b::4]:41040, peer=131
Added time data, samples 80, offset +0 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
ResendWalletTransactions()
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=132
Added time data, samples 81, offset +0 (+0 minutes)
nTimeOffset = +0
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=133
Added time data, samples 82, offset +0 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
connect() to 37.187.96.183:45328 failed after select(): Connection refused (111)
connected to self at 172.17.0.28:35320, disconnecting
ProcessSpork : got updated spork 10007 (SPORK_8_MASTERNODE_PAYMENT_ENFORCEMENT) with value 4070908800 (signed at 1610238601)
Wrote spork SPORK_8_MASTERNODE_PAYMENT_ENFORCEMENT to database
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=136
Added time data, samples 83, offset +0 (+0 minutes)
nTimeOffset = +0
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
socket recv error Connection reset by peer (104)
receive version message: /Sap Core:1.3.3/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=137
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
receive version message: /SAPP Core:1.3.3.2/: version 70926, blocks=901815, us=176.9.3.85:45328, peer=138
Added time data, samples 84, offset +0 (+0 minutes)
asking peer for sporks
AdvertiseLocal: advertising address 176.9.3.85:45328
Segmentation fault (core dumped)
```

**Comment by bedri** [comment]
This issue is solved with version 1.4.1
Closing the issue.
